doc/releases/changelog-0.35.0.md

```diff
@@ -43,13 +173,9 @@
 
   Previously, mid-circuit measurements would be automatically replaced with an additional qubit
   using the `@qml.defer_measurements` transform, so the above circuit would have required thousands
-  of qubits to simulate.
-
-* The `default.qubit` device now supports mid-circuit measurements without using the deferred measurement
-  principle when using finite shots. The native execution mode comprises all features supported with `qml.defer_measurements`, including
-  classical control, collecting statistics, and post-selection. This PR notably introduces support for
-  post-selection, measurement value lists, and composite measurements.
-  [(#5120)](https://github.com/PennyLaneAI/pennylane/pull/5120)
+  of qubits to simulate. All the mid-circuit measurement features supported through deferred
+  measurements, such as post-selection and classical control, are now supported natively with
+  a finite number of shots in `default.qubit`.
 
 <h4>Work easily and efficiently with operators 🔧</h4>
 
```

Comment:
> line 191: can we say that we "renamed" the Pauli operators and that the old names will be deprecated soon?

I added a note here: 16c0a598275c671c53d77bdc201ce77f6ce15a5e. We decided to keep the `PauliX`, etc., names for the foreseeable future just because of how ingrained their use is.

> line 200: I'm not sure that I understand the feature 🤔

Yes maybe this is a more subtle one :thinking: The idea is that if you do `op = qml.X(0) + qml.Y(0)`, there are two things: (i) the frontend `op` which is a PennyLane operator, and (ii) the backend Pauli representation. If we do arithmetic with `op`, then we can use the Pauli representation to speed up calculations. The rest of that bullet point just talks about the `PauliSentence` and `PauliWord` objects which form the Pauli representation. Maybe a tweak of the first sentence could help? E.g., "Operators in PennyLane can have a backend Pauli representation, which can be used to perform faster operator arithmetic. Now, the Pauli representation will be automatically used for calculations when available." @Qottmann is this still accurate?

> line 244: is there a way to reword this such that it makes the impact on the user clearer?

How about "The `Sum` and `Prod` classes, two major components of our new approach to operator arithmetic, have been updated to reach feature parity with `Hamiltonian` and `Tensor`, respectively. This should minimize the effort to port over any existing code."

> line 269: is that all we have to say about the clifford device? All good if so, just wondering

:thinking: This one was a bit more lightweight as it's a frontend for another package (`stim`), but if I were to expand on anything it would be showing an example with errors present in the circuit. You think that would be good?

Reply:
> "Operators in PennyLane can have a backend Pauli representation, which can be used to perform faster operator arithmetic. Now, the Pauli representation will be automatically used for calculations when available." @Qottmann is this still accurate?

👍 (there are still more tweaks to do but overall thats the idea yeah)